### PR TITLE
pull config from pull request when a pr is labeled rather than the default branch

### PR DESCRIPTION
**Pull request by PaliC**
Addresses https://github.com/pytorch/test-infra/issues/3849

Here we use the [oktokit api](https://github.com/probot/octokit-plugin-config) to get the config file from the branch that the pr is on.

One minor security issue here is that it is generally not recommended that the bot reads configs / files from the non default branch as it could be a security risk. However, since we are doing this for labels (and using that flow), it should be low risk.

The nock unit tests were changed a little bit to look for configs from the branch of the pr rather than the default branch. Those unit tests should be enough to show that this works.

### CODE REVIEW

File: torchci/lib/bot/ciflowPushTrigger.ts [hunk 1]
```diff
@@ -136,7 +141,16 @@ async function handleLabelEvent(
...
   }
-  const config: any = await tracker.loadConfig(context);
+  if (repoKey(context) == "pytorch/pytorch-canary"){
+    const config: any =  await Octokit.config.get({
+      owner: `${context.repo().owner}`,
+      repo: `${context.repo().repo}`,
+      path: ".github/pytorch-probot.yml",
+      branch: `${context.payload.ref}`,
+    });
+  }else{
```
Comment (ZainRizvi) [inline torchci/lib/bot/ciflowPushTrigger.ts:151]: What effect does this have? (i.e. how is this config different than the non-canary version?)

File: torchci/lib/bot/ciflowPushTrigger.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { Context, Probot } from "probot";
-import { CachedConfigTracker, isPyTorchPyTorch } from "./utils";
+import { CachedConfigTracker, isPyTorchPyTorch, repoKey } from "./utils";
```
Comment (huydhn) [inline torchci/lib/bot/ciflowPushTrigger.ts:2]: Nit: I don't see `repoKey` is used anywhere, so it could be removed?

**Review by huydhn — APPROVED** [review]
LGTM!

File: torchci/lib/bot/ciflowPushTrigger.ts (hunk 2 above)
Comment (PaliC) [reply torchci/lib/bot/ciflowPushTrigger.ts:2]: thanks for the catch :)

**Pull request merged**